cabal-install/Distribution/Client/Setup.hs

```diff
@@ -836,7 +847,12 @@ runCommand = CommandUI {
          "Builds and then runs the specified executable. If no executable is "
       ++ "specified, but the package contains just one executable, that one "
       ++ "is built and executed.\n",
-    commandNotes        = Nothing,
+    commandNotes        = Just $ \pname ->
+          "Examples:\n"
+       ++ "  " ++ pname ++ " run\n"
+       ++ "    Run the only executable in the current package;\n"
+       ++ "  " ++ pname ++ " run foo -- --key=value\n"
+       ++ "    Run the `foo` executable with some flag.\n",
```

Comment:
This is a slightly hard one to explain. I'm slightly worried that people might think this is saying that the `--key=value` is a thing cabal run takes, rather than passing anything extra directly to the program to be run.